### PR TITLE
chore: automate changelog generation on release branch

### PR DESCRIPTION
**Pull request by gastonfournier**
## About the changes
This generates the changelog in the branch taking into account latest release (it's conditional to wether the release ends with .0 or not)

### CODE REVIEW

File: .github/workflows/publish-new-version.yaml
Comment (gastonfournier) [inline .github/workflows/publish-new-version.yaml:24]: Hopefully with permissions: contents: write we don't need this anymore

**Comment by chriswk** [comment]
Having tested this, it seems to work

**Review by chriswk — APPROVED** [review]
LGTM. But I was part of the work so we might want to wait for another approval

**Comment by gastonfournier** [comment]
> LGTM. But I was part of the work so we might want to wait for another approval

We'll test it out with @andreas-unleash in 5.5.4 patch release

**Pull request merged**
